daemons/ipa-kdb/ipa_kdb_principals.c

```diff
@@ -333,6 +333,11 @@ static enum ipadb_user_auth ipadb_get_user_auth(struct ipadb_context *ipactx,
     if (gcfg != NULL)
         gua = gcfg->user_auth;
 
+    /* lcontext == NULL means ipadb_get_global_config() failed to load
+     * global config and cleared the ipactx */
+    if (ipactx->lcontext == NULL)
+        return IPADB_USER_AUTH_NONE;
```

Comment:
What implications does this have? IIUC if the LDAP connection has dropped then this authentication will happen as password-only. Should it just fail outright instead of falling back to a perhaps less-secure mechanism (that may well fail)?

Reply:
We default to `IPADB_USER_AUTH_NONE` even in the global auth configuration. See `ipadb_parse_user_auth()` -- in case our global configuration is missing, we use `IPADB_USER_AUTH_NONE`. I think we should keep this.